File: src/uk/co/ruben9922/numbergame/NumberTile.java
package uk.co.ruben9922.numbergame;

class NumberTile extends Tile {
    private int number;

    public NumberTile(Colour colour, int number) {
        super(colour);
        this.number = number;
    }

    public int getNumber() {
        return number;
    }

    @Override
    public String toString() {
        return colour.toString() + " " + number;
    }

    public TileType getTileType() {
        return TileType.NUMBER;
    }
}

File: NumberTileGame/NumberTile.cs
namespace NumberTileGame
{
    internal class NumberTile : Tile
    {
        public int Number { get; set; }

        public const int MinValue = 1;
        public const int MaxValue = 13;

        public NumberTile(Colour colour, int number) : base(colour)
        {
            Number = number;
        }

        public override string ToString()
        {
            return $"NumberTileGame.NumberTile(Number={Number},Colour={Colour})";
        }
    }
}